src/iceberg/table_metadata.cc

```diff
@@ -153,14 +154,70 @@ Result<MetadataFileCodecType> TableMetadataUtil::CodecFromFileName(
   return MetadataFileCodecType::kNone;
 }
 
+class GZipDecompressor {
```

Comment:
Should we just define it as a function because it is just an one-off thing? Perhaps declare it in `gzip_internal.h` so we can test it individually?

Reply:
ok, considering of future use, i think move those code to "gzip_internal.cc" to utils dir is much more data.